### ISSUE TITLE
Split DataStore2 functionality, make current saving process optional

### ISSUE THREAD

**Issue by Kampfkarren** [Kampfkarren/Roblox#37]
DataStore2 is, at its core, a generic and more easy to use data store API that caches. I'd like everyone to use DataStore2.

However, the saving method of DataStore2 prevents this vision. Furthermore, if Roblox ever does anything to prevent this saving method, developers are at a complete loss.

Thus, a generic internal data store saving API should be added, and offer both the berezaa method and normal data store saving. Berezaa should be the default unless anything changes in the future.

This should NOT break any games, and nobody should have to change anything.

**Comment by visskiss** [comment]
What is the “method” that Datastore2 uses. Doesn’t it just use the built in Roblox data store methods?

**Comment by Kampfkarren** [comment]
At the end of the day, it does. However, this is the exact way it does it:

https://github.com/Kampfkarren/Roblox/blob/master/DataStore2.module.lua#L29-L37

**Comment by visskiss** [comment]
It seems like every save is growing the data store size. Is that right?

**Comment by Kampfkarren** [comment]
Yeah, but Roblox seems to have yet to care--several top games use this and berezaa's games in specific were using it for years most likely.

**Comment by visskiss** [comment]
It doesn’t seem like a smart strategy. What about trimming older entries? Say remove all but that last four entries. This guarantees a back up, but is also space efficient.

**Comment by Kampfkarren** [comment]
It's been suggested before and I believe the rationale against it aside from ordered data stores in general have really unforgiving rate limits, so trimming off any requests we can has large gains.

**Comment by visskiss** [comment]
After going for a run and thinking about it, I recall the google rationale for immutable data is that requires far less computing power. It’s also less prone to error so I guess it probably makes sense. Does caching mean that updates to the actual data store are relatively rare, like when a user quits?

**Comment by Kampfkarren** [comment]
Yeah, DataStore2 only saves when either the player leaves or you call `:Save()`. Makes data store gets aside from the very first instant, and makes setting always instant.
